samza-core/src/main/java/org/apache/samza/config/TaskConfig.java

```diff
@@ -0,0 +1,298 @@
+        .filter(StringUtils::isNotBlank)
+        .map(checkpointManagerFactoryName -> Util.getObj(checkpointManagerFactoryName, CheckpointManagerFactory.class)
+            .getCheckpointManager(this, metricsRegistry));
+  }
+
+  /**
+   * Get the systemStreamPartitions of the broadcast stream. Specifying
+   * one partition for one stream or a range of the partitions for one
+   * stream is allowed.
+   *
+   * @return a Set of SystemStreamPartitions
+   */
+  public Set<SystemStreamPartition> getBroadcastSystemStreamPartitions() {
+    HashSet<SystemStreamPartition> systemStreamPartitionSet = new HashSet<>();
+    List<String> systemStreamPartitions = getList(BROADCAST_INPUT_STREAMS, Collections.emptyList());
+
+    for (String systemStreamPartition : systemStreamPartitions) {
+      int hashPosition = systemStreamPartition.indexOf("#");
+      if (hashPosition == -1) {
+        throw new IllegalArgumentException("incorrect format in " + systemStreamPartition
+            + ". Broadcast stream names should be in the form 'system.stream#partitionId' or 'system.stream#[partitionN-partitionM]'");
+      } else {
+        String systemStreamName = systemStreamPartition.substring(0, hashPosition);
+        String partitionSegment = systemStreamPartition.substring(hashPosition + 1);
+        SystemStream systemStream = StreamUtil.getSystemStreamFromNames(systemStreamName);
+
+        if (Pattern.matches(BROADCAST_STREAM_PATTERN, partitionSegment)) {
+          systemStreamPartitionSet.add(new SystemStreamPartition(systemStream, new Partition(Integer.valueOf(partitionSegment))));
+        } else {
+          if (Pattern.matches(BROADCAST_STREAM_RANGE_PATTERN, partitionSegment)) {
+            int partitionStart = Integer.valueOf(partitionSegment.substring(1, partitionSegment.lastIndexOf("-")));
+            int partitionEnd = Integer.valueOf(partitionSegment.substring(partitionSegment.lastIndexOf("-") + 1, partitionSegment.indexOf("]")));
+            if (partitionStart > partitionEnd) {
+              LOGGER.warn("The starting partition in stream " + systemStream.toString() + " is bigger than the ending Partition. No partition is added");
+            }
+            for (int i = partitionStart; i <= partitionEnd; i++) {
+              systemStreamPartitionSet.add(new SystemStreamPartition(systemStream, new Partition(i)));
+            }
+          } else {
+            throw new IllegalArgumentException("incorrect format in " + systemStreamPartition
+                + ". Broadcast stream names should be in the form 'system.stream#partitionId' or 'system.stream#[partitionN-partitionM]'");
+          }
+        }
+      }
+    }
+    return systemStreamPartitionSet;
+  }
+
+  /**
+   * Get the SystemStreams for the configured broadcast streams.
+   *
+   * @return the set of SystemStreams for which there are broadcast stream SSPs configured.
+   */
+  public Set<SystemStream> getBroadcastSystemStreams() {
```

Comment:
Up to you if you want to do this now, but I'd prefer to do this at the call sites and remove this method. I'd prefer Config classes to have one way of doing things, and ideally be logic-less. Multiple similar methods cause confusion.

Reply:
Case in point: getInputStreams vs getAllInputStreams below.